enhancements/windows-containers/windows-machine-config-operator.md

```diff
@@ -0,0 +1,185 @@
+authors:
+  - "@ravisantoshgudimetla"
+  - "@aravindhp"
+reviewers:
+  - "@sdodson"
+  - "@derekwaynecarr"
+  - "@openshift/openshift-team-cloud"
+  - "@openshift/openshift-team-mco"
+approvers:
+  - "@sdodson"
+  - "@derekwaynecarr"
+creation-date: 2020-05-28
+last-updated: 2020-05-28
+status: implementable
+replaces:
+  - "/enhancements/windows-containers/ansible-dev-preview.md"
+---
+
+# Windows Machine Config Operator
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The intent of this enhancement is to allow a cluster administrator to add a
+Windows compute node as a day 2 operation with a prescribed configuration to an
+installer provisioned OpenShift 4.6 cluster and enable scheduling of Windows
+workloads. The targeted platforms for 4.6 are vSphere, Azure and AWS. The
+cluster has to be configured with Hybrid OVN networking.
+
+## Motivation
+
+The main motivation behind this enhancement is to satisfy customer
+requirement of being able to run Windows workloads on OpenShift clusters.
+
+### Goals
+
+As part of this enhancement we plan to do the following:
+* Provide workflows for adding and removing Windows instances provisioned using
+  the Machine API to OpenShift clusters
+* Perform all the required steps within the VM for it to be added to the cluster
+  as an OpenShift worker node and removed from it.
+
+### Non-Goals
+
+As part of this enhancement we do not plan to support:
+* Installing the container runtime in the Windows node
+* Upgrades (This will be addressed in a separate enhancement)
+  * Windows operating system upgrades
+  * Node management like k8s compute component upgrades and reboots
+* OpenShift Builds
+* OpenShift DeploymentConfigs
+
+## Proposal
+
+The two main pieces that will allow us to achieve our goals is the Windows
+Machine Config operator (WMCO) and a Windows Machine Config Bootstrapper (WMCB)
+binary. WMCO will collect information from the cluster and transfer it to the
+Windows node. The WMCB will then perform the necessary steps on the node for it
+to join the cluster. [WMCB](https://github.com/openshift/windows-machine-config-bootstrapper)
+has already been developed as part of the Windows Containers
+[dev preview](https://github.com/openshift/windows-machine-config-bootstrapper/blob/master/tools/ansible/docs/ocp-4-4-with-windows-server.md).
+
+### Justification
+
+The reason we are using a WMCB binary rather than a container image is
+that all Windows container images are packaged with a Windows kernel and Red Hat
+has a policy to not ship 3rd party kernels for support reasons.
+
+### Design Details
+
+The WMCO image will be packaged with the following binaries:
```

Comment:
Everything below are windows binaries, right? Just want to make it clear these are artifacts that we copy to windows hosts.

Reply:
Correct, they are all Windows binaries that we copy to the Windows VM.